components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java

```diff
@@ -175,6 +176,48 @@ public SecretResponse getSecret(String secretType, String name) {
         }
     }
 
+    /**
+     * To make a partial update or update the specific property of the secret.
+     *
+     * @param name               secret name.
+     * @param secretPatchRequest secret's patch details.
+     * @return Updated secret.
+     */
+    public SecretResponse patchSecret(String secretType, String name, SecretPatchRequest secretPatchRequest) {
+
+        Secret secret, responseDTO;
+        try {
+            secret = SecretManagementServiceHolder.getSecretConfigManager().getSecret(secretType, name);
+            if (secret == null) {
+                throw handleException(Response.Status.NOT_FOUND, SecretManagementConstants.ErrorMessage.
+                                ERROR_CODE_SECRET_NOT_FOUND, name);
+            }
+            String path = secretPatchRequest.getPath();
+            SecretPatchRequest.OperationEnum operation = secretPatchRequest.getOperation();
+            //Only the Replace operation supported with PATCH request
+            if (SecretPatchRequest.OperationEnum.REPLACE.equals(operation)) {
+                if (SecretManagementConstants.VALUE_PATH.equals(path)) {
+                    secret.setSecretValue(secretPatchRequest.getValue());
+                } else if (SecretManagementConstants.DESCRIPTION_PATH.equals(path)) {
+                    secret.setDescription(secretPatchRequest.getValue());
+                } else {
+                    throw handleException(Response.Status.BAD_REQUEST, SecretManagementConstants.ErrorMessage
+                            .ERROR_CODE_INVALID_INPUT, "Path");
+                }
+                responseDTO = SecretManagementServiceHolder.getSecretConfigManager().replaceSecret(secretType, secret);
```

Comment:
This will update the complete secret entry in the DAO. So can we have 2 different DAO and service methods to update only value and description?

Reply:
Resolved